ONSdigital/census-rh-service
# Cr 67 handle postal fulfilment requests

# Motivation and Context

This change is required so that census-rh-service is using the latest version of census-int-common-service, which provides the GenericEvent, Header, and DateTimeUtil classes (amongst other things).

# What has changed
The pom has been changed to point to version 0.0.19 of the census-int-common-service. Other changes have been made only to fix broken import statements, and so on, as a result of using the new version.





# How to test?
This be tested simply by running the census-rh-service in the usual way. Make sure that any timestamps end in either Z or in +0000 (ideally test both formats as both should be accepted) e.g. use the following instructions:

Use the readme file to firstly set up Google credentials locally. Then run Rabbitmq locally using the command 'docker-compose up -d'. Run the census-rh-service from the command line as follows:
mvn spring-boot:run

The RabbitMQ Management console should now show the following five exchanges:
	1.	Case-deadletter-exchange
	2.	Events
	3.	Respondent-deadletter-exchange
	4.	Respondent-outbound-exchange
	5.	Uac-deadletter-exchange

Set up the events exchange, in the ‘Bindings’ section, so that it has the following two bindings defined:
	1.	To Case.Gateway, Routing key is event.case.lifecycle
	2.	To UAC.Gateway, Routing key is event.uac.updates

Create an environment variable to hold the name of your Google Cloud Platform project, using the export command at the terminal (in your census-rh-service repo): export GOOGLE_CLOUD_PROJECT = “your-project-name” e.g. census-rh-ellacook1

Use the appropriate JSON format (using the types of timestamp format that you would like to test) to put a CaseUpdated type message on the events exchange. You can then run the code and should see the message appear in Google Cloud Platform within a bucket named your-project-name-case NB. To see the buckets you need to select Storage --> Browser (within GCP). The code will create the bucket if it does not already exist.
